crates/resolver/src/async_std.rs

```diff
@@ -0,0 +1,178 @@
+//! async-std connection provider.
+
+use std::{future::Future, io};
+
+use crate::proto::{async_std::AsyncStdRuntimeProvider, runtime::Executor};
+use crate::{
+    config::{NameServerConfig, ResolverOpts},
+    name_server::{ConnectionProvider, GenericConnector},
+};
+
+/// async-std connection provider.
+#[derive(Clone, Default)]
+pub struct AsyncStdConnectionProvider {
+    runtime_provider: AsyncStdRuntimeProvider,
+    connection_provider: GenericConnector<AsyncStdRuntimeProvider>,
+}
+
+impl Executor for AsyncStdConnectionProvider {
+    fn new() -> Self {
+        let p = AsyncStdRuntimeProvider::new();
+        Self {
+            runtime_provider: p,
+            connection_provider: GenericConnector::new(p),
+        }
+    }
+
+    fn block_on<F: Future>(&mut self, future: F) -> F::Output {
+        self.runtime_provider.block_on(future)
+    }
+}
+
+impl ConnectionProvider for AsyncStdConnectionProvider {
+    type Conn = <GenericConnector<AsyncStdRuntimeProvider> as ConnectionProvider>::Conn;
+    type FutureConn = <GenericConnector<AsyncStdRuntimeProvider> as ConnectionProvider>::FutureConn;
+    type RuntimeProvider = AsyncStdRuntimeProvider;
+
+    fn new_connection(
+        &self,
+        config: &NameServerConfig,
+        options: &ResolverOpts,
+    ) -> Result<Self::FutureConn, io::Error> {
+        self.connection_provider.new_connection(config, options)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use test_support::subscribe;
+
+    use super::AsyncStdConnectionProvider;
+    #[cfg(feature = "system-config")]
+    use crate::testing::{hosts_lookup_test, system_lookup_test};
+    use crate::{
+        config::ResolverConfig,
+        proto::runtime::Executor,
+        testing::{
```

Comment:
We'll need to either enable the `testing` feature in a dev-dependency or remove the `testing` feature entirely, since it was only needed for `async-std-resolver tests`.

Reply:
Removed the feature, forgot about that yesterday, thanks!